java/xgboost4j/src/main/java/org/dmlc/xgboost4j/util/Trainer.java

```diff
@@ -0,0 +1,230 @@
+/*
+ Copyright (c) 2014 by Contributors 
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+    
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+package org.dmlc.xgboost4j.util;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.dmlc.xgboost4j.IEvaluation;
+import org.dmlc.xgboost4j.Booster;
+import org.dmlc.xgboost4j.DMatrix;
+import org.dmlc.xgboost4j.IObjective;
+
+
+/**
+ * trainer for xgboost
+ * @author hzx
+ */
+public class Trainer {
+    private static final Log logger = LogFactory.getLog(Trainer.class);
+    
+    /**
+     * Train a booster with given parameters.
+     * @param params Booster params.
+     * @param dtrain Data to be trained.
+     * @param round Number of boosting iterations.
+     * @param evalMats Data to be evaluated (may include dtrain)
+     * @param evalNames name of data (used for evaluation info)
+     * @param obj customized objective (set to null if not used)
+     * @param eval customized evaluation (set to null if not used)
+     * @return trained booster
+     */
+    public static Booster train(Params params, DMatrix dtrain, int round,
```

Comment:
user Iterable<Entry<String, String> > for params?
Ideally it should be Iterable <Entry<String, anytype that have toString> >